test/core.js

```diff
@@ -337,6 +337,42 @@ describe('snabbdom', function() {
         assert.strictEqual(elm.childNodes[0].nodeType, 1);
         assert.strictEqual(elm.childNodes[0].textContent, 'Hello');
       })
+      it('can work with domApi', function () {
+        var domApi = {
+            tagName: function(elm) { return elm.tagName.toUpperCase(); },
+            getTextContent: function(elm) { return elm.text; },
+            isElement: function(elm) { return !!elm.tagName; },
+            isText: function(elm) { return !!elm.text; }
+        };
+        var Elm = function(tagName) {
```

Comment:
I don't think u need `Elm` constructor, just create a DOM structure like described here: 

https://github.com/snabbdom/snabbdom/blob/master/test/core.js#L268-L273

Then call `toVNode()` on that structure and analyze the returned vnode structure to make sure that it is correct.

you can provide a domApi where tagName returns a custom tagname for any element processed, e.g.:

```js
tagName: function(elm) { return 'X-' + elm.tagName.toUpperCase(); },
```

which means that both, the parent a child vnodes should have a `sel` that contains the `x-` prefix.